Source/SwiftLintFramework/Rules/AttributesRule.swift

```diff
@@ -0,0 +1,385 @@
+            violations = validateKind(file,
+                                      attributeShouldBeOnSameLine: attributeShouldBeOnSameLine,
+                                      dictionary: dictionary)
+        } else {
+            violations = []
+        }
+
+        return validateFile(file, dictionary: dictionary) + violations
+    }
+
+    private func validateTestableImport(file: File) -> [StyleViolation] {
+        let pattern = "@testable[\n]+\\s*import"
+        let excludingKinds = SyntaxKind.commentAndStringKinds()
+        return file.matchPattern(pattern, excludingSyntaxKinds: excludingKinds).map {
+            let match = file.contents.substringWithByteRange(start: $0.location, length: $0.length)
+            let location = (match?.lastIndexOf("import") ?? 0) + $0.location
+
+            return StyleViolation(ruleDescription: self.dynamicType.description,
+                                  severity: configuration.severityConfiguration.severity,
+                                  location: Location(file: file, byteOffset: location))
+        }
+    }
+
+    private func validateKind(file: File,
+                              attributeShouldBeOnSameLine: Bool,
+                              dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        let attributes = parseAttributes(dictionary)
+
+        guard !attributes.isEmpty,
+            let offset = (dictionary["key.offset"] as? Int64).flatMap({ Int($0) }),
+            let (lineNumber, _) = file.contents.lineAndCharacterForByteOffset(offset) else {
+            return []
+        }
+
+        let line = file.lines[lineNumber - 1]
+
+        let tokens = file.syntaxMap.tokensIn(line.byteRange)
+        let attributesTokensWithRanges = tokens.flatMap { attributeName($0, file: file) }
+
+        let attributesTokens = Set(attributesTokensWithRanges.map { $0.0 })
+        var isViolation = false
+
+        do {
+            let previousAttributesWithParameters = try attributesFromPreviousLines(lineNumber - 1,
+                                                                                   file: file)
+            let previousAttributes = Set(previousAttributesWithParameters.map { $0.0 })
+
+            let alwaysInSameLineAttributes = configuration.alwaysInSameLine
+            let alwaysInNewLineAttributes =
+                createAlwaysInNewLineAttributes(previousAttributesWithParameters,
+                                                attributesTokens: attributesTokensWithRanges,
+                                                line: line, file: file)
+
+            if !attributesTokens.intersect(alwaysInNewLineAttributes).isEmpty {
+                isViolation = true
+            } else if !previousAttributes.intersect(alwaysInSameLineAttributes).isEmpty {
+                isViolation = true
+            } else {
+
+                // ignore whitelisted attributes
+                let operation: Set<String> -> Set<String> -> Set<String> =
```

Comment:
I couldn't find a better name 😅
Maybe we should drop this clever way and just have an if with two paths?